ocs_ci/deployment/aws.py

```diff
@@ -291,15 +291,24 @@ class OCPDeployment(BaseOCPDeployment):
         def __init__(self):
             super(AWSUPI.OCPDeployment, self).__init__()
             upi_repo_name = f'openshift-misc-{config.RUN["run_id"]}'
+            self.ocp_version = ".".join(
+                config.DEPLOYMENT.get('installer_version').split('.')[:-2]
+            )
+
             self.upi_repo_path = os.path.join(
                 constants.EXTERNAL_DIR, upi_repo_name,
             )
 
-            self.upi_script_path = os.path.join(
-                self.upi_repo_path,
-                'v3-launch-templates/functionality-testing'
-                '/aos-4_2/hosts/'
-            )
+            if self.ocp_version == '4.3':
```

Comment:
Instead of having separate constants for each version could we make this a bit more simple and just use the ocp version to construct the filepath directly?

```python
upi_common_base = 'v3-launch-templates/functionality-testing'
self.upi_script_path = os.path.join(
    self.upi_repo_path,
    upi_common_base,
    self.ocp_version.replace('.','_'),
    'hosts'
)
```
If we _do_ want to go with setting explicit paths for certain versions, I would suggest we use a dict instead of individual constants. This way we don't need an `if`/`elif` block for each separate version and instead we can do it once. This also has the benefit of scaling better when we inevitably have more versions to account for.

```python
upi_common_base = 'v3-launch-templates/functionality-testing'
ocp_version_paths = {
    '4.2': os.path.join(upi_common_base, 'aos-4_2/hosts/'),
    '4.3': os.path.join(upi_common_base, 'aos-4_3/hosts/')
}
self.upi_script_path = os.path.join(
    self.upi_repo_path,
    ocp_version_paths[self.ocp_version],
)
```